src/app/shared/services/dialog-confirmation.service.spec.ts

```diff
@@ -0,0 +1,231 @@
+import { DialogService } from './dialog.service'
+import { anyString, instance, mock, verify, when } from '@typestrong/ts-mockito'
+import { RundownService } from '../../core/abstractions/rundown.service'
+import { BasicRundownStateService } from '../../core/services/basic-rundown-state.service'
+import { Observable, of } from 'rxjs'
+import { TestEntityFactory } from '../../test/factories/test-entity.factory'
+import { RundownMode } from '../../core/enums/rundown-mode'
+import { Rundown } from '../../core/models/rundown'
+import { DialogConfirmationService } from './dialog-confirmation.service'
+
+describe(DialogConfirmationService.name, () => {
+  it('should be created', () => {
+    const testee: DialogConfirmationService = createTestee()
+
+    expect(testee).toBeTruthy()
+  })
+
+  describe(DialogConfirmationService.prototype.switchRehearsalRundownDialog.name, () => {
+    it('should show dialog when all rundowns are idle when going for rehearsal', () => {
+      const testEntityFactory: TestEntityFactory = new TestEntityFactory()
+
+      const mockedRunownService = mock<RundownService>()
+
+      const mockedSubscription: Observable<void> = mock<Observable<void>>()
+      when(mockedRunownService.rehearse(anyString())).thenReturn(instance(mockedSubscription))
+
+      const mockedDialogService: DialogService = createDialogServiceMock()
+
+      const mockedBasicRundoStateService = mock<BasicRundownStateService>()
+      when(mockedBasicRundoStateService.getNonIdleRundown()).thenReturn(undefined)
+
+      const testee: DialogConfirmationService = createTestee({
+        mockedBasicRundownStateService: mockedBasicRundoStateService,
+        mockedRundownService: mockedRunownService,
+        mockedDialogService: mockedDialogService,
+      })
+      testee.switchRehearsalRundownDialog(testEntityFactory.createRundown({ id: 'this', mode: RundownMode.INACTIVE }))
+
+      verify(mockedDialogService.createConfirmDialog).once()
+    })
+
+    it('should show dialog when another rundown is already in active mode when going for rehearsal', () => {
+      const testEntityFactory: TestEntityFactory = new TestEntityFactory()
+
+      const mockedRunownService = mock<RundownService>()
+
+      const mockedRehearseSubscription: Observable<void> = mock<Observable<void>>()
```

Comment:
To create a test that resembles "real-life" as close as possible, we should not mock value objects like observables. You can use the `EMPTY` observable exposed by RxJS for this purpose.